### DISCUSSION TITLE
How to schedule a job with a partitioned asset?

### DISCUSSION THREAD

**Discussion by garethbrickman** [dagster-io/dagster#24531] [Q&A] [answered]
I want to schedule an asset job for a partitioned asset so that the partition for the then current day can be materialized.

**Comment by garethbrickman** [comment, marked as answer]
There's a special purpose method for that: [build_schedule_from_partitioned_job](https://docs.dagster.io/_apidocs/schedules-sensors#dagster.build_schedule_from_partitioned_job). There's a code example included in that documentation that I've copied below: 

```python
from dagster import (
    DailyPartitionsDefinition,
    asset,
    build_schedule_from_partitioned_job,
    define_asset_job,
    Definitions,
)

@asset(partitions_def=DailyPartitionsDefinition(start_date="2020-01-01"))
def asset1():
    ...

asset1_job = define_asset_job("asset1_job", selection=[asset1])

# The created schedule will fire daily
asset1_job_schedule = build_schedule_from_partitioned_job(asset1_job)

defs = Definitions(assets=[asset1], schedules=[asset1_job_schedule])
```

In the beloiw screenshot you can see when testing the schedule evaluation for a given day in the future that the `dagster/partition` tag will be for that day's date:
![Screenshot 2024-09-17 at 09 43 47](https://github.com/user-attachments/assets/70b8f6c2-c11e-4560-96a6-beedc1383ad6)

There's an additional example linked in our [Partitioning assets documentation here.](https://docs.dagster.io/concepts/partitions-schedules-sensors/partitioning-assets#examples)

**Reply by v1gnesh** [reply]
I may be mistaken, but isn't Dagster trying to move away from needing to define jobs & schedules "manually" like this?
